docs/snippets/filter/expected_error.cpp

```diff
@@ -0,0 +1,8 @@
+struct Error { std::string what; };
+fn::expected<int, Error> ex{12};
+
+auto value = ex 
+    | filter([](auto &&i) { return i >= 42; }, 
+             [](auto) { return Error{"Less than 42"}; });
+
+REQUIRE(value.error().what == "Less than 42");
```

Comment:
I noticed you have all the small files for (I assume) demonstrating compiler behavior. 
Just in case, wanted to ask if you know that in other cases you can extract sub snippets from larger files using a variety of approaches.

Reply:
I have not tried to extract `@code` blocks from doxygen if that's what you are referring to. Does that work with Znai? What are the other ways?

Even if it's possible though, personally i'm not a fan of adding code examples into comments in the actual codebase - there is enough code in there as it is. We can adjust if others disagree but imo small snippet files are the better choice seeing that we are essentially writing most of the documentation by hand anyway (as opposed to generating the entire doc doxygen style).

Reply:
Not a fan of adding code examples to comments as well. Was wondering if you saw this: https://testingisdocumenting.org/znai/snippets/snippets-manipulation